### DISCUSSION TITLE
Do plugins made for the first-party (Windows) app work?

### DISCUSSION THREAD

**Discussion by anthonyma94** [StreamController/StreamController#132] [Q&A]
Hi, very excited that this project is being developed. I see that plugins are a prominent feature, does this mean that all existing plugins for the stream deck will work, or does it have to be written specifically for the Stream controller?

**Comment by Core447** [comment]
StreamController has its own store with its own plugins. Compatibility with the Windows plugins might be theoretically possible, but most really nice Windows plugins need a very tight OS integration that will never really work on Linux. And instead of spending hundreds of hours to make such a system okayish, I prefer to invest the time in new cool features for StreamController and/or in new native plugins.

**Reply by anthonyma94** [reply]
Thanks for the response! I understand that not all plugins will work, especially ones interacting with the OS. However, many plugins are just wrappers communicating with a third-party API (like this [home assistant plugin](https://github.com/cgiesche/streamdeck-homeassistant)). Having some kind of support for these would dramatically increase the app's adoption rate imo.

**Reply by Core447** [reply]
Yeah, the problem is that, as I said, it would be very time consuming to develop a translation layer. We already have equivalents of many popular Windows plugins (among others one for home assistant). However, if someone is willing to dig into this, I have no problem helping.

**Comment by Cfomodz** [comment]
**No, Windows Stream Deck plugins do not work with StreamController.**


Use StreamController's native plugin store (built into the app). Many popular Windows plugins already have Linux equivalents, including:
- Home Assistant
- API-based integrations
- Media controls
- System commands

## For Developers

Want a specific plugin? Create a native version using the [StreamController Wiki](https://streamcontroller.github.io/docs) or request it on [Discord](https://discord.gg/MSyHM8TN3u).
